### PR TITLE
SNOW-737313 Set default parameter chunk_size in create_dataframe/write_pandas

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-737313

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Set `chunk_size` 100,000 by default in stored procedures.

### CODE REVIEW

File: src/snowflake/snowpark/session.py [hunk 1]
```diff
@@ -1330,6 +1330,7 @@ def create_dataframe(
         self,
         data: Union[List, Tuple, "pandas.DataFrame"],
         schema: Optional[Union[StructType, List[str]]] = None,
+        chunk_size: Optional[int] = 100000,
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/session.py:1333]: I don't like adding parameter of a helper function to the caller's signature. Should we just not expose this parameter?  The only concern is `chunk_size` is not the size of the file to be uploaded, but the number of rows in a chunk, so not exposing this parameter means the issue reported could still occur with this "fix".

**Comment by sfc-gh-sfan** [comment]
I think this change should be for sp connector only. I'm not sure if client in general needs this.

**Comment by sfc-gh-stan** [comment]
> I think this change should be for sp connector only. I'm not sure if client in general needs this.
> 
> You might want to move your change to the sp connector repo

I see, that makes sense. Closing this.

**Comment by sfc-gh-stan** [comment]
Actually, we still need some change in this repo, since `Session.create_dataframe` calls `Session.write_pandas` where `chunk_size` is default to `None`, which overwrites the default value in SP connector.

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -1187,7 +1187,7 @@ def write_pandas(
         *,
         database: Optional[str] = None,
         schema: Optional[str] = None,
-        chunk_size: Optional[int] = None,
+        chunk_size: Optional[int] = 100000,
```
[the remark this answers is not in the document]
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/session.py:1190]: Oh I almost forgot about `is_in_stored_procedure()`, this is perfect 👍

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**